src/controllers/graphql/resolvers/post/postResolver.js

```diff
@@ -1,28 +1,110 @@
-const { AuthenticationError } = require('apollo-server');
 const db = require('../../connectors');
-
 const Post = db.sequelize.models.post;
-const ReplyService = require('../../../../services/replyService');
 const PostService = require('../../../../services/postService');
 const { FETCH_TOP_X_POSTS } = require('../../../../utils/constants');
+const {
+  AuthenticationError,
+  gql,
+  makeExecutableSchema,
+} = require('apollo-server');
+
+const typeDefs = gql`
+  scalar Date
+
+  type Query {
+    getAllPosts: [Post]
+    getPostById(id: ID!): Post
+    getFollowerPosts: [Post]
+    getHundredxPosts: [Post]
+    # getTextPosts: [Post]
+    # getNewsPosts: [Post]
+    # getVideoPosts: [Post]
+    # getTrendingPosts(trending: String): [Post]
+    # getPostsByTags(tags: [String]): [Post]
+  }
+
+  type Mutation {
+    createPost(post: Post!): Post
+    updatePost(post: Post!): Post
+    deletePost(id: ID!): boolean # Return boolean if delete was successful or not
+  }
+
+  type Post {
+    id: ID!
+    title: String
+    body: String
+    authorId: String
+    # Other post values
+  }
+`;
 
-module.exports = {
+const resolvers = {
   Query: {
     getAllPosts: async (_, args) => {
-      return await Post.findAll({order: [['hot', 'ASC']], limit: FETCH_TOP_X_POSTS});
+      return await Post.findAll({
+        order: [['hot', 'ASC']],
+        limit: FETCH_TOP_X_POSTS
+      });
     },
-    getPost: async (_,args) => {
+
+    getPost: async (_, args) => {
       return await Post.findById(args.postId);
     },
+
     getFollowerPosts: async (_, args, { authenticatedUserInstance }) => {
-      return !authenticatedUserInstance
-        ? new AuthenticationError('ERROR_GETTING_FOLLOWER_POSTS')
-        : await authenticatedUserInstance.getMyFollowersPosts();
+      if (!authenticatedUserInstance) {
+        throw new AuthenticationError('INVALID_USER');
+      }
+      return await authenticatedUserInstance.getMyFollowersPosts();
     },
+
     getHundredxPosts: async (_, args) => {
       return await PostService.fetchHundredxResteemedPosts();
     }
+
+    // getTextPosts: async(_, args) => {
```

Comment:
commented queries are the ones needing implementation

Reply:
ok